### ISSUE TITLE
Unexpected result from BinaryCrossEntropy loss calculation in keras fit/evaluate

### ISSUE THREAD

**Issue by will-cern** [keras-team/tf-keras#404]
Hello,

I am opening this issue here in response to the potential bug described here: https://discuss.tensorflow.org/t/how-is-binarycrossentropy-loss-calculated-in-keras-fit-evaluate/11331/11  The reproducer code is as follows (in tf 2.8.2):

```
from tensorflow import keras
import numpy as np
loss_func = keras.losses.BinaryCrossentropy()
nn = keras.Sequential([
  keras.layers.Dense(2**8, input_shape=(1,), activation='relu'),
  keras.layers.Dense(2,activation='softmax')
])
nn.compile(loss=loss_func,optimizer='adam')
train_x = np.array([0.4]) # this is an arbitrary input
train_y = np.array([[1.,0.]])
train_q = nn.predict(train_x)
print("train_q = ",train_q)
print("Evaluated loss = ",nn.evaluate(train_x,train_y))
print("Function loss = ",loss_func(train_y,train_q).numpy())
print("Manual loss = ", -np.log(train_q[0,0]) )
```

The last two ways of calculating the loss agree and are what I would expect, but the value returned by the `evaluate` method is different and unexplainable for me. 

Any help in debugging this issue would be much appreciated.

Thanks
Will
